src/main/java/uk/gov/hmcts/reform/pip/data/management/database/ArtefactRepository.java

```diff
@@ -1,13 +1,31 @@
 package uk.gov.hmcts.reform.pip.data.management.database;
 
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 import uk.gov.hmcts.reform.pip.data.management.models.publication.Artefact;
 
+import java.time.LocalDateTime;
+import java.util.List;
 import java.util.Optional;
+import java.util.UUID;
 
 @Repository
 public interface ArtefactRepository extends JpaRepository<Artefact, Long> {
     Optional<Artefact> findBySourceArtefactIdAndProvenance(String sourceArtefactId, String provenance);
 
+    Optional<Artefact> findByArtefactId(UUID artefactId);
+
+    @Query(value = "select * from Artefact where search->'court-id'->>0 = :searchVal and display_from < "
```

Comment:
Just been testing it all out. Display from and display to are optional fields, therefore if they are not set no results are returned.

I'm not sure what the expected behaviour for no display from / to is, so we'll need to double check this.

Reply:
This will now be captured in a seperate ticket